components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java

```diff
@@ -244,6 +244,107 @@ public Response subscriptionsPost(SubscriptionDTO body, String xWSO2Tenant, Mess
+            if (body.getApiId() != null) {
+                if (!RestAPIStoreUtils.isUserAccessAllowedForAPIByUUID(body.getApiId(), tenantDomain)) {
+                    RestApiUtil.handleAuthorizationFailure(RestApiConstants.RESOURCE_API, body.getApiId(), log);
+                }
+            }  else {
+                RestApiUtil.handleBadRequest(
+                        "Request must contain either apiIdentifier or apiProductIdentifier and the relevant type", log);
+                return null;
+            }
+
+            Application application = apiConsumer.getApplicationByUUID(applicationId);
+            if (application == null) {
+                //required application not found
+                RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_APPLICATION, applicationId, log);
+                return null;
+            }
+
+            if (!RestAPIStoreUtils.isUserAccessAllowedForApplication(application)) {
+                //application access failure occurred
+                RestApiUtil.handleAuthorizationFailure(RestApiConstants.RESOURCE_APPLICATION, applicationId, log);
+            }
+
+            ApiTypeWrapper apiTypeWrapper = apiConsumer.getAPIorAPIProductByUUID(body.getApiId(), tenantDomain);
+
+            //Validation for allowed throttling tiers and Tenant based validation for subscription. If failed this will
+            //  throw an APIMgtAuthorizationFailedException with the reason as the message
+            RestAPIStoreUtils.checkSubscriptionAllowed(apiTypeWrapper, body.getThrottlingPolicy());
+
+            apiTypeWrapper.setTier(body.getThrottlingPolicy());
+
+            SubscriptionResponse subscriptionResponse = apiConsumer
+                    .updateSubscription(apiTypeWrapper, username, application.getId(), subscriptionId,
+                            currentThrottlingPolicy, requestedThrottlingPolicy);
+            SubscribedAPI addedSubscribedAPI = apiConsumer
+                    .getSubscriptionByUUID(subscriptionResponse.getSubscriptionUUID());
+            SubscriptionDTO addedSubscriptionDTO = SubscriptionMappingUtil.fromSubscriptionToDTO(addedSubscribedAPI);
+            WorkflowResponse workflowResponse = subscriptionResponse.getWorkflowResponse();
+            if (workflowResponse instanceof HttpWorkflowResponse) {
+                String payload = workflowResponse.getJSONPayload();
+                addedSubscriptionDTO.setRedirectionParams(payload);
+            }
+
+            return Response.ok(new URI(RestApiConstants.RESOURCE_PATH_SUBSCRIPTIONS + "/" +
+                    addedSubscribedAPI.getUUID())).entity(addedSubscriptionDTO).build();
+
+        } catch (APIMgtAuthorizationFailedException e) {
+            //this occurs when the api:application:tier mapping is not allowed. The reason for the message is taken from
+            // the message of the exception e
+            RestApiUtil.handleAuthorizationFailure(e.getMessage(), e, log);
```

Comment:
use exceptionCodes

Reply:
It seems this APIMgtAuthorizationFailedException is handled like same as this in all other usages.